src/Chapter3.hs

```diff
@@ -971,6 +1088,33 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek 
+  = Monday
+  | Tuesday
+  | Wednesday
+  | Thursday
+  | Friday
+  | Saturday
+  | Sunday 
+  deriving(Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend day = 
+  if day == Saturday || day == Sunday
+    then True
+    else False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day = 
+  if day == Sunday
```

Comment:
You could also use the instance of `Bounded` in here:

```suggestion
    if day == maxBound then minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂